polygon.js

```diff
@@ -0,0 +1,216 @@
+const PI                   = Math.PI,
+      NUMBER_OF_POLYGON    = 6,
+      MAX_NUMBER_OF_ANGLES = 7,
+      length               = 100;   
+
+function main() {
+    var canvas   = document.getElementById("canvas"),
+        ctx      = canvas.getContext('2d'),
+        polygons = [],
+        mouse = {
+            x: 0,
+            y: 0
+        }
+    canvas.setAttribute('height',window.innerHeight - 20);
+    canvas.setAttribute('width',window.innerWidth - 20);
+    polygons = initPolygon(NUMBER_OF_POLYGON, MAX_NUMBER_OF_ANGLES);
+    colorDraw(polygons, canvas, ctx);
+    canvas.addEventListener('mousedown', function(evt){
+        mouse.x = evt.clientX;
+        mouse.y = evt.clientY;
+        polygons.forEach(function(item, i, arr){
+            arr[i].selected = inPolygon(mouse.x, mouse.y, item);
+        });
+    });
+    canvas.addEventListener('mousemove', function(evt){
+        polygons.forEach(function(item, i, arr){
+            if(item.selected){
+                for (let j = 0; j < item.x.length; j++) {
+                    item.x[j] += evt.clientX - mouse.x;
+                    item.y[j] += evt.clientY - mouse.y;
+                }
+                mouse.x = evt.clientX;
+                mouse.y = evt.clientY;
+                colorDraw(polygons, canvas, ctx);
+            }
+        });
+    });
+    canvas.addEventListener('mouseup', function(evt){
+        mouse.x = 0;
+        mouse.y = 0;
+        polygons.forEach(function(item, i, arr){
+            if(item.selected){
+                for(let j = 0; j < arr.length; j++){
+                    if(i == j) continue
+                    if(intOfPol(item, arr[j]) || overlappingOfPolygons(item, arr[j]) || overlappingOfPolygons(arr[j], item)){
```

Comment:
shouldn't `overlappingOfPolygons(item, arr[j])` and `overlappingOfPolygons(arr[j], item)` should return same value?